conda/cli/conda_argparse.py

```diff
@@ -1845,7 +1845,12 @@ def add_parser_channels(p):
         action="append",
         choices=["jlap", "lock"],
         help="jlap: Download incremental package index data from repodata.jlap; implies 'lock'. "
-        "lock: use locking when reading, updating index (repodata.json) cache. ",
+        "lock: use locking when reading, updating index (repodata.json) cache. Now enabled.",
+    )
+    channel_customization_options.add_argument(
+        "--no-lock",
```

Comment:
Since this is a lock for the repodata handling, I wonder if something more specific be more appropriate like

```suggestion
        "--no-lock-repodata",
```
or

```suggestion
        "--no-repodata-lock",
```

This might prevent any conflict in case we add other locking in the future..